airbyte-cdk/python/pyproject.toml

```diff
@@ -1,8 +1,91 @@
-# Defines Python build system settings.
-[build-system]
-requires = [
-    "setuptools>=42",
-    "wheel"
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+[tool.poetry]
+name = "airbyte_cdk"
+description = "A framework for writing Airbyte Connectors."
+authors = ["Airbyte <contact@airbyte.io>"]
+license = "MIT"
+readme = "README.md"
+keywords = ["airbyte", "connector-development-kit", "cdk"]
+documentation = "https://docs.airbyte.io/"
+repository = "https://github.com/airbytehq/airbyte"
+
+# The version of the airbyte-cdk package is used at runtime to validate manifests. That validation must be
+# updated if our semver format changes such as using release candidate versions.
+version = "0.54.0"
+
+exclude = ["unit_tests"]
+include = [
+    "airbyte_cdk",
+    "airbyte_cdk/py.typed",
+    "airbyte_cdk/sources/declarative/declarative_component_schema.yaml"
+]
+
+[tool.poetry.urls]
+"Tracker" = "https://github.com/airbytehq/airbyte/issues"
+
+[tool.poetry.dependencies]
+python = ">=3.8.1,<4.0"
+airbyte-protocol-models = "0.4.2"
+backoff = "^2.2.1"
+dpath = ">=2.0.1,<2.1.0"
+isodate = ">=0.6.1,<0.7.0"
+jsonschema = ">=3.2.0,<3.3.0"
+jsonref = ">=0.2,<1.0"
+genson = "1.2.2"
+pydantic = ">=1.10.8,<2.0.0"
+pyrate-limiter = ">=3.1.0,<3.2.0"
+pyyaml = ">=6.0.1"
+deprecated = ">=1.2,<2.0"
+jinja2 = ">=3.1.2,<3.2.0"
+wcmatch = "8.4"
+pendulum = "^2.1.2"
+requests = "^2.31.0"
+requests-cache = "^1.1.1"
+cachetools = "^5.3.2"
+
+# Sphinx Docs Extras:
+sphinx = {version = ">=4.2,<5.0", optional = true}
+sphinx-rtd-theme = {version = ">=1.0,<2.0", optional = true}
+
+# File-based Extras:
+avro = {version = ">=1.11.2,<1.12.0", optional = true}
+fastavro = {version = ">=1.8.0,<1.9.0", optional = true}
+pyarrow = {version = "12.0.1", optional = true}
+
+# File-based Extras (Unstructured Docs):
+markdown = {version = "^3.5.1", optional = true}
+pdf2image = {version = "1.16.3", optional = true}
+pdfminer-six = {version = "20221105", optional = true}
+pytesseract = {version = "0.3.10", optional = true}
+unstructured = {version = "0.10.27", extras = ["file-based"]} # Unstructured version can't be bumped higher due to transitive dependencies we can't provide:
```

Comment:
why are extras called out in the dependency for these two, but specified in `tool.poetry.extras` for the others?

Reply:
Good catch. I experimented with both formats, and missed getting them all consistent.

I believe both formats are (basically?) identical - either declaring the name of the extra inline with the  dependency or declaring the group of dependencies that comprise the extra below as an array of dependent names.

I'll fix so they all use the same convention.